### ISSUE TITLE
Support taproot

### ISSUE THREAD

**Issue by acoutts** [inapay/bech32#5]
It appears that trying to decode a taproot address does not work, saying the checksum has failed.

**Comment by dspicher** [comment]
Indeed, the checksum is the only thing that has changed for the `bech32m` address scheme used for taproot:
https://github.com/bitcoin/bips/blob/master/bip-0350.mediawiki#Bech32m
